src/Simulation/TargetDefinitions/Decompositions/MResetXWithNoReuse.qs

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Measurement {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Diagnostics;
+
+    /// # Summary
+    /// Measures a single qubit in the X basis,
+    /// and resets it to a fixed initial state
+    /// following the measurement.
+    ///
+    /// # Description
+    /// Performs a single-qubit measurement in the $X$-basis,
+    /// and ensures that the qubit is returned to $\ket{0}$
+    /// following the measurement.
+    ///
+    /// # Input
+    /// ## target
+    /// A single qubit to be measured.
+    ///
+    /// # Output
+    /// The result of measuring `target` in the Pauli $X$ basis.
+    @EnableTestingViaName("Test.TargetDefinitions.MResetX")
+    operation MResetX (target : Qubit) : Result {
+        return Measure([PauliX], [target]);
```

Comment:
For each of these "WithNoReuse" variants, the platform in question doesn't support reusing qubits after measurement, so a reset after measurement is not necessary. Further, calling `Reset` might trigger an error because it might internally use the qubit, which would count as a use of after measurement. 

I kept the comments the same to match the existing documentation of the method, but I think it's an interesting point to ask: should the doc comments on target packages always match the default or shift depending on the underlying target behavior?

Reply:
I might add a non-doc comment just to explain to someone reading the code why the code and doc comments don't match.

I think having different doc comments for different implementations of the same class would break the docs system :-).